contracts/ServiceRegistry.sol

```diff
@@ -226,7 +226,9 @@ contract ServiceRegistry is Initializable, ERC2771RecipientUpgradeable, UUPSUpgr
      * @param _platformId platform ID on which the Service token was minted
      * @param _serviceDataUri token Id to IPFS URI mapping
      */
-    function createOpenServiceFromBuyer(uint256 _platformId, string calldata _serviceDataUri) public returns (uint256) {
+    function createOpenServiceFromBuyer(uint256 _platformId, string calldata _serviceDataUri) public payable returns (uint256) {
+        uint16 servicePostingFee = talentLayerPlatformIdContract.getServicePostingFee(_platformId);
+        require(msg.value == servicePostingFee, "Non-matching funds");
         talentLayerPlatformIdContract.isValid(_platformId);
```

Comment:
now we got two external calls here, that's not optimal, we can probably do the same with only one

Reply:
we can do a global getPlatform probably, can you check the gas impact please (now and after getPlatform implemented)

Reply:
Actually "isValid" check is already done  in getServicePostingFee
So can be removed